### PR TITLE
Donor_Details_Clean - seperate changes from old branch

### PR DESCRIPTION
**Pull request by anyajain5**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [ ] I have performed a self-review and reviewed the changed files
- [ ] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [x] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [x] No sensitive information committed (API keys, passwords, etc.)
- [x] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Summary by CodeRabbit

* **New Features**
  * Item-level controls when viewing a request: search, status multiselect, sort by date or quantity with asc/desc, and list/gallery view toggle.
  * Clickable gallery cards for request items showing image (or placeholder), category, quantity, date, and status badges.
  * New donation request details panel with donor contact, address, and notes; Back control resets item filters.
* **Changes**
  * Request lists now omit fully completed requests and support donor/name filtering and sorting.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Hack4Impact-UMD/journey-home/pull/90?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: components/donation-requests/DRDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { DonationRequest } from "@/types/donations";
+
+function SectionHeader({ title }: { title: string }) {
+    return (
+        <div className="h-10 bg-[#FAFAFB] border-light-border border flex items-center font-bold text-sm text-primary shrink-0">
```
Comment (constrictingsnake) [inline components/donation-requests/DRDetails.tsx]: the left line accent is a bit longer than it's supposed to be

File: components/donation-requests/DRDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <span className="border-l-2 border-primary px-4">{title}</span>
+        </div>
+    );
+}
+
+function DetailRow({ label, value }: { label: string; value: string }) {
+    return (
+        <div className="min-h-10 border-light-border border-b border-x flex items-center text-sm text-text-1">
+            <span className="w-[40%] px-4 font-bold shrink-0">{label}</span>
```
Comment (constrictingsnake) [inline components/donation-requests/DRDetails.tsx]: the label needs to be the same color as the section header

File: components/donation-requests/DRContentsTable.tsx
Comment (constrictingsnake) [inline components/donation-requests/DRContentsTable.tsx]: h-full pushes the drdetails component to the bottom, requiring a scroll, something that we don't want. removing h-full and modifying the spacing between the list view and details would match the figma better

File: components/donation-requests/DRContentsGallery.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+function DRGalleryCard({
+    item,
+    onOpen,
+}: {
+    item: DonationItem;
+    onOpen: () => void;
+}) {
+    return (
+        <div
+            className="border border-[#E1E1E1] rounded-[4px] bg-white hover:bg-blue-50 cursor-pointer flex flex-col px-2.5 pt-2.5 pb-3 gap-[10px]"
```
Comment (constrictingsnake) [inline components/donation-requests/DRContentsGallery.tsx]: pt-2.5 and pt-3 is innacurate, the figma says 12px on all sides

File: components/donation-requests/DRContentsGallery.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                        alt={item.item.name}
+                        className="w-full h-full object-cover rounded-[4px]"
+                    />
+                ) : (
+                    <span className="text-xs text-gray-400">No image</span>
+                )}
+            </div>
+
+            <div className="flex flex-col gap-1">
+                <span className="text-sm font-bold text-text-1 truncate">
```
Comment (constrictingsnake) [inline components/donation-requests/DRContentsGallery.tsx]: the font should bne semi bold

File: app/donation-requests/new/page.tsx [hunk 5]
```diff
@@ -42,33 +108,111 @@ export default function DonationRequestsPage() {
...
+                                </button>
+                            ))}
+                        </div>
+                    )}
                 </div>
 
-                <span className="font-bold py-4.5">
-                    {selectedDR.donor.firstName} {selectedDR.donor.lastName}
-                </span>
+                <div className="ml-auto flex items-center gap-1">
```
Comment (constrictingsnake) [inline app/donation-requests/new/page.tsx]: these buttons should align with the figma more, there's no border needed and rather than the color becoming blue on click it should just become a darker shade

File: app/donation-requests/new/page.tsx [hunk 6]
```diff
@@ -42,33 +108,111 @@ export default function DonationRequestsPage() {
...
+                        className={`w-8 h-8 flex items-center justify-center border border-light-border ${viewMode === "gallery" ? "bg-[#FAFAFB] text-primary" : "bg-white text-gray-400"}`}
+                        onClick={() => setViewMode("gallery")}
+                    >
+                        <GridIcon active={viewMode === "gallery"} />
+                    </button>
+                </div>
             </div>
-            <div className="flex-1 overflow-auto min-h-0">
-                <DRContentsTable request={selectedDR} openItem={setSelectedItem} />
+
```
Comment (constrictingsnake) [inline app/donation-requests/new/page.tsx:135]: there should be more spacing between the search bar/filters and the actual content

File: components/donation-requests/DRContentsTable.tsx
Comment (constrictingsnake) [inline components/donation-requests/DRContentsTable.tsx]: the accent on the left of the categories are taller than the figma

File: components/donation-requests/DRContentsGallery.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                        item={item}
+                        onOpen={() => openItem(item)}
+                    />
+                ))}
+            </div>
+        </div>
+    );
+}
+
+function DRGalleryCard({
```
Comment (constrictingsnake) [inline components/donation-requests/DRContentsGallery.tsx:24]: the checkbox functionality needs to be translated over to the gallery view as well - as of rn, someone should be able to select multiple items

**Review by constrictingsnake — CHANGES_REQUESTED** [review]

File: components/donation-requests/DRDetails.tsx (hunk 2 above)
Comment (anyajain5) [reply components/donation-requests/DRDetails.tsx]: I wasn't totally sure what color it should be? Should I change it from black to teal, bc in the figma it was black.

**Review by joelchem — APPROVED** [review]
